### PR TITLE
Remove previous SR4R form and wizard

### PR DESCRIPTION
**Pull request by steventux**
## Context

We've released the redesigned reasons for rejection workflow and will soon remove the associated feature flag.

## Changes proposed in this pull request

Remove routes, controller, wizard, views and i18n error messages for the previous iteration of the reasons for rejection flow.

## Link to Trello card

https://trello.com/c/U5KJQGNE/5049-remove-previous-r4r-form-and-wizard

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/shared/rejection_reasons/reasons_for_rejection_component.html.erb [hunk 1]
```diff
@@ -192,70 +154,44 @@
...
   <div class="app-rejection">
-    <%= content_tag(subheading_tag_name, 'Future applications', class: 'govuk-heading-s') %>
+    <h3 class="govuk-heading-s">Future applications</h3>
     <p class="govuk-body">
       <%= I18n.t(
         "reasons_for_rejection.interested_in_future_applications.reason.#{reasons.interested_in_future_applications_y_n.downcase}",
         provider_name: application_choice.provider.name,
       ) %>
     </p>
-    <% if editable? %>
```
Comment (cwrw) [inline app/components/shared/rejection_reasons/reasons_for_rejection_component.html.erb:255]: could we also remove `editable?` from `app/components/shared/rejection_reasons/rejection_reasons_component.html.erb`

File: app/components/shared/rejection_reasons/structured_rejection_reasons_component.rb [hunk 2]
```diff
@@ -20,10 +20,6 @@ def paragraphs(input)
     input.split("\r\n")
   end
 
-  def subheading_tag_name
```
Comment (cwrw) [inline app/components/shared/rejection_reasons/structured_rejection_reasons_component.rb:23]: could we also remove the editable method and the editable attribute?

File: app/components/shared/rejection_reasons/reasons_for_rejection_component.rb [hunk 3]
```diff
@@ -2,4 +2,8 @@
 # This component class supports the rendering of rejection reasons from the initial iteration of structured rejection reasons.
 # See https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/docs/reasons-for-rejection.md
 #
-class RejectionReasons::ReasonsForRejectionComponent < RejectionReasons::StructuredRejectionReasonsComponent; end
+class RejectionReasons::ReasonsForRejectionComponent < RejectionReasons::StructuredRejectionReasonsComponent
+  def editable?
```
Comment (cwrw) [inline app/components/shared/rejection_reasons/reasons_for_rejection_component.rb:6]: do we need to still maintain an editable method?

File: spec/mailers/previews/candidate_mailer_preview.rb [hunk 4]
```diff
@@ -561,10 +533,6 @@ def feedback_received_for_application_rejected_by_default_apply_again(reasons_tr
     CandidateMailer.feedback_received_for_application_rejected_by_default(application_choice, show_apply_again_guidance)
   end
 
-  def feedback_received_for_application_rejected_by_default_apply_again_redesigned_reasons
-    feedback_received_for_application_rejected_by_default_apply_again(:with_redesigned_rejection_reasons)
-  end
-
```
Comment (cwrw) [inline spec/mailers/previews/candidate_mailer_preview.rb:567]: could we also remove `reasons_for_rejection_with_qualifications` and `reasons_type` methods?

**Review by cwrw — COMMENTED** [review]
looks good 🧹  just some minor comments

File: app/components/shared/rejection_reasons/reasons_for_rejection_component.html.erb (hunk 1 above)
Comment (steventux) [reply app/components/shared/rejection_reasons/reasons_for_rejection_component.html.erb:255]: @cwrw  `editable` is used here https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/views/provider_interface/rejections/check.html.erb#L19 but I think it makes sense to move the `editable?` method into the component specific to the current iteration of R4R, it's not used by the other components now, will update.

Comment (steventux) [reply app/components/shared/rejection_reasons/reasons_for_rejection_component.html.erb:255]: https://github.com/DFE-Digital/apply-for-teacher-training/pull/6876/commits/e0e3d09934b286605a2edc0ff98e31d4a332cb61 should refactor the scope of what is `editable?`

**Review by cwrw — APPROVED** [review]
🌟

**Pull request merged**